packages/dds/merge-tree/src/test/obliterate.partialLength.spec.ts

```diff
@@ -0,0 +1,313 @@
+            );
+            const localObliterateOp = client.obliterateRangeLocal(
+                0,
+                "hello ".length,
+            );
+
+            validatePartialLengths(localClientId, client.mergeTree, [
+                { seq: refSeq, len: "hello world".length, localSeq: refSeq },
+                { seq: refSeq + 1, len: "world".length, localSeq: refSeq + 1 },
+                { seq: refSeq + 2, len: "world".length, localSeq: refSeq + 2 },
+            ]);
+
+            client.applyMsg(client.makeOpMessage(localObliterateOp, refSeq + 2));
+
+            validatePartialLengths(remoteClientId, client.mergeTree, [
+                { seq: refSeq, len: "hello world".length },
+                { seq: refSeq + 1, len: "hello world".length },
+                { seq: refSeq + 2, len: "world".length },
+            ]);
+        });
+
+        it("passes for remote remove and remote obliterate", () => {
+            client.removeRangeRemote(
+                0,
+                "hello ".length,
+                refSeq + 1,
+                refSeq,
+                client.getLongClientId(remoteClientId),
+            );
+            client.obliterateRange({
+                start: 0,
+                end: "hello ".length,
+                refSeq,
+                clientId: remoteClientId + 1,
+                seq: refSeq + 2,
+                overwrite: false,
+                opArgs: undefined as any,
+            });
+
+            validatePartialLengths(localClientId, client.mergeTree, [
+                { seq: refSeq, len: "hello world".length },
+                { seq: refSeq + 1, len: "hello world".length },
+                { seq: refSeq + 2, len: "world".length },
+            ]);
+            validatePartialLengths(remoteClientId, client.mergeTree, [
+                { seq: refSeq, len: "hello world".length },
+                { seq: refSeq + 1, len: "hello world".length },
+                { seq: refSeq + 2, len: "world".length },
+            ], 0);
+            validatePartialLengths(remoteClientId + 1, client.mergeTree, [
+                { seq: refSeq, len: "hello world".length },
+                { seq: refSeq + 1, len: "hello world".length },
+                { seq: refSeq + 2, len: "world".length },
+            ], 0);
+        });
+    });
+
+    describe("overlapping obliterate+obliterate", () => {
+        it("passes for local obliterate and remote obliterate", () => {
+            const localObliterateOp = client.obliterateRangeLocal(
+                0,
+                "hello ".length,
+            );
+            client.obliterateRange({
+                start: 0,
+                end: "hello ".length,
+                refSeq,
+                clientId: remoteClientId,
+                seq: refSeq + 1,
+                overwrite: false,
+                opArgs: undefined as any,
+            });
+
+            validatePartialLengths(localClientId, client.mergeTree, [
```

Comment:
FYI: I'm not sure all of these intermediate validations are correct. At this point in time, the partial lengths should only have actual ops for `seq: refSeq` and `seq: refSeq + 1`. However they should also be able to compute the length at `{ seq: refSeq, localSeq: 0 }` and `{ seq: refSeq + 1, localSeq: 1 }` in case `localObliterateOp` needs to be rebased to have refSeq `refSeq + 1` (this happens if `client` disconnects and reconnects after the remote client's obliterate is applied). So having `refSeq + 2` in here doesn't seem right. In general the set of perspectives that partial lengths is able to compute the block length for is actually the cartesian product of the ranges `{ seq: minSeq }...{ seq: currentSeq }` and `{ localSeq: oldestUnackedLocalSeq }...{ localSeq: mostRecentLocalSeq }`

I don't think this is critical for checking the tests in (especially as they're skipped) so I'm going to approve, but you should be aware of it once you start looking to enable them.